src/osd/OSD.cc

```diff
@@ -6539,10 +6643,13 @@ void OSD::handle_osd_map(MOSDMap *m)
 
   if (m->newest_map && m->newest_map > last) {
     dout(10) << " msg say newest map is " << m->newest_map << ", requesting more" << dendl;
-    osdmap_subscribe(osdmap->get_epoch()+1, true);
+    osdmap_subscribe(osdmap->get_epoch()+1, false);
   }
-  else if (is_booting()) {
-    start_boot();  // retry
+  else if (is_preboot()) {
+    if (m->get_source().is_mon())
+      _preboot(m->oldest_map, m->newest_map);
```

Comment:
This doesn't work. The message's oldest_map is set based on the maps the OSD already has, not necessarily the oldest map on the monitor. "send_latest(op, m->sb.current_epoch+1);"